server/mqtt.go

```diff
@@ -4160,7 +4160,25 @@ func mqttSubscribeTrace(pi uint16, filters []*mqttFilter) string {
 // message and this is the callback for a QoS1+ subscription because in that
 // case, it will be handled by the other callback. This avoid getting duplicate
 // deliveries.
+func mqttDoNotDeliverSubject(subject string) bool {
```

Comment:
Not sure if this function should be under this comment (that belongs to the QoS0 callback). But more importantly, that may not work for JS Api prefix that use domain, so would it be simpler to just reject anything that starts with `$JS.` and `$MQTT.`?
We could also maybe optimize with something like (maybe we need to check that subject len > 0, but I don't think we would be invoked in a callback with an empty subject)
```
if subject[0] != '$' {
    return false
}
if strings.HasPrefix(subject, "$JS.") || strings.HasPrefix(subject, mqttJSARepliesPrefix) {
    return true
}
return false
```

Reply:
RE: Comment - oops

RE: Optimize - sure, SG.

Reply:
![image](https://github.com/nats-io/nats-server/assets/1187448/fbb012a0-7e4a-4cfa-9ce0-a55abaece0a7)

Reply:
(oops, left in `-race` there, but the relative perf. is similar, but MUCH faster of course)